lib/linux_stats/os/sys_block_stat.rb

```diff
@@ -37,57 +40,65 @@ module Column
   end
 
   module DataFile
-    CPUINFO = '/proc/cpuinfo'
-    DISK_STATS = '/proc/diskstats'
-    SECTOR_SIZE = '/sys/block/sda/queue/hw_sector_size'
+    CPUINFO = '/cpuinfo'
+    DISK_STATS = '/diskstats'
+    SECTOR_SIZE = '/block/sda/queue/hw_sector_size'
   end
 
-  def self.cpuinfo
-    ret = 0
-    IO.readlines(DataFile::CPUINFO).each do |line|
-      ret += 1 if line =~ /^processor/
+  class Reporter
+    def initialize(_data = nil, proc_data_directory = PROC_DIRECTORY_DEFAULT, sys_data_directory = SYS_DIRECTORY_DEFAULT)
+      set_data_paths(proc_data_directory,sys_data_directory)
+
+      @bytes_per_sector = sector_size
+      @ignore_disks = [
+          '^dm-[0-9]',
+          '^fd[0-9]',
+          '^ram',
+          '^loop',
+          '^sr',
+          '^sd.*[0-9]'
+      ]
+      @num_cpu = cpuinfo
+      @watched_disks_list = watched_disks
+      set_stats
     end
-    ret
-  end
-
-  def self.sector_size
-    begin
-      return File.read(DataFile::SECTOR_SIZE).strip.to_i
-    rescue
-      # handle CentOS 5
-      return 512
+    
+    def set_data_paths(proc_data_directory = nil, sys_data_directory = nil)
+      @sys_directory = sys_data_directory
+      @proc_cpuinfo_source = "#{proc_data_directory}#{DataFile::CPUINFO}"
+      @proc_diskstats_source = "#{proc_data_directory}#{DataFile::DISK_STATS}"
+      @sys_sectorsize_source = "#{sys_data_directory}#{DataFile::SECTOR_SIZE}"
     end
-  end
 
-  def self.watched_disks(data = nil)
-    disk_list = []
-    data = File.read(DataFile::DISK_STATS) unless data
-    data.each_line do |line|
-      words = line.split
-      disk_name = words[2]
-      disk_list.push disk_name if File.exists? "/sys/block/#{disk_name}/stat"
-    end
-    IGNORE_DISKS.each do |pattern|
-      disk_list.reject! { |x| x =~ /#{pattern}/ }
+    def cpuinfo
+      ret = 0
+      IO.readlines(@proc_cpuinfo_source).each do |line|
+        ret += 1 if line =~ /^processor/
+      end
+      ret
     end
-    disk_list
-  end
 
-  BYTES_PER_SECTOR = sector_size
-  IGNORE_DISKS = [
-    '^dm-[0-9]',
-    '^fd[0-9]',
-    '^ram',
-    '^loop',
-    '^sr',
-    '^sd.*[0-9]'
-  ]
-  NUM_CPU = cpuinfo
-  WATCHED_DISKS = watched_disks
+    def sector_size
+      begin
+        return File.read(@sys_sectorsize_source).strip.to_i
+      rescue
+        # handle CentOS 5
+        return 512
+      end
+    end
 
-  class Reporter
-    def initialize(_data = nil)
-      set_stats
+    def watched_disks(data = nil)
+      disk_list = []
+      data = File.read(@proc_diskstats_source) unless data
+      data.each_line do |line|
+        words = line.split
+        disk_name = words[2]
```

Comment:
That was part of the original code, unchanged here (other than position in file).  Lines from /proc/diskstats look like "8       5 sda5 244 0 2200 568 0 0 0 0 0 568 568", so after reading a line and doing the split, the device name 2nd index location.